### ISSUE TITLE
False positive: eq_op consts

### ISSUE THREAD

**Issue by alexheretic** [rust-lang-nursery/rust-clippy#2420]
`eq_op` warning is not useful for some const maths.

```rust
const CONFIG_DISTANCE_METERS: f64 = 100.0;
const CONFIG_SECONDS_TO_COVER_DISTANCE: f64 = 100.0;
const CONFIG_SPEED: f64 = CONFIG_DISTANCE_METERS / CONFIG_SECONDS_TO_COVER_DISTANCE;
```
```
3 | const CONFIG_SPEED: f64 = CONFIG_DISTANCE_METERS / CONFIG_SECONDS_TO_COVER_DISTANCE;
  |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: #[warn(eq_op)] on by default
  = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.184/index.html#eq_op
```

Here this warning only makes sense if the same _named_ constant is used on either side of the operand. Otherwise this warning is just telling me that one constant happens to be the same as another.
